statistics/handle/handle_hist.go

```diff
@@ -46,14 +46,14 @@ type StatsLoad struct {
 	SubCtxs        []sessionctx.Context
 	NeededItemsCh  chan *NeededItemTask
 	TimeoutItemsCh chan *NeededItemTask
-	WorkingColMap  map[model.TableItemID][]chan model.TableItemID
+	WorkingColMap  map[model.TableItemID][]chan stmtctx.StatsLoadResult
 }
 
 // NeededItemTask represents one needed column/indices with expire time.
 type NeededItemTask struct {
-	TableItemID model.TableItemID
-	ToTimeout   time.Time
-	ResultCh    chan model.TableItemID
+	TableItemResult stmtctx.StatsLoadResult
```

Comment:
It's a little strange that task contains result, is TableItemID enough? And wrap it to be TableItemResult when write response.